bears/vcs/git/GitRevertInspectBear.py

```diff
@@ -0,0 +1,149 @@
+import re
+
+from difflib import SequenceMatcher
+
+from bears.vcs.git.GitCommitMetadataBear import GitCommitMetadataBear
+from coalib.bears.GlobalBear import GlobalBear
+from coalib.misc.Shell import run_shell_command
+from coalib.results.Result import Result
+
+
+class GitRevertInspectBear(GlobalBear):
+    LANGUAGES = {'Git'}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+
+    BEAR_DEPS = {GitCommitMetadataBear}
+
+    GIT_REVERT_COMMIT_RE = re.compile(
+        r'Revert\s\".+\"\n\nThis\sreverts\scommit\s([0-9a-f]{40})\.')
+
+    def check_modified_file_similarity(self, file_path, reverted_commit_sha):
+        """
+        Compare the changes in file modified by the
+        revert commit with the changes actually
+        expected in the revert commit.
+
+        :param file_path:           Relative path to the modified file.
+        :param reverted_commit_sha: Commit HASH of reverted commit.
+        """
+        with open(file_name, 'r') as f:
+            revert_file_content = f.read()
+
+        head_commit_branch = run_shell_command(
+            'git rev-parse --abbrev-ref HEAD')[0].strip('\n')
+        switch_branch_command = ('git checkout ' +
+                                 head_commit_branch)
+
+        create_new_branch_command = (
+            'git checkout -b inspectrevertbranch HEAD^')
+        run_shell_command(create_new_branch_command)
+
+        create_expected_revert_commit = ('git revert %s --no-'
+                                         'edit' %
+                                         reverted_commit_sha)
+        _, err = run_shell_command(create_expected_revert_commit)
+        if err:
+            self.warn('Cannot compare the modified files.')
+            run_shell_command('git revert --abort')
+            run_shell_command(switch_branch_command)
+            run_shell_command('git branch -D inspectrevertbranch')
+            return
+
+        expected_revert_commit_sha = run_shell_command(
+            'git rev-parse HEAD')[0].strip('\n')
+        files_modified = ('git show --pretty="" --name-status %s' %
+                          (expected_revert_commit_sha))
+        files_modified_list = []
+        for line in files_modified:
+            pos = line.find('\t')
+            change = line[:pos]
+            if change == 'M':
+                files_modified_list.append(line[pos+1:])
+
+        if file_name in files_modified_list:
```

Comment:
I think we don't need to add this check again, as it's already done in line 142

Reply:
These files are of the "expected revert commit" that we are creating for comparison.
Line 142 checks files in the reverted commit. So as to check any extra added or deleted files.

Reply:
If the file doesn't exist in the "expected revert commit", it means that the file doesn't exist in the reverted commit either (since the "expected revert commit" is created purely with git revert without any modification, it should be the negate of the reverted commit). And this function probably won't be called in another source code, so I think it's safe enough to remove this check here